metadata/metadata.go

```diff
@@ -0,0 +1,130 @@
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"github.com/mackerelio/mackerel-agent/config"
+	"github.com/mackerelio/mackerel-agent/logging"
+)
+
+var logger = logging.GetLogger("metadata")
+
+// Generator generates metadata
+type Generator struct {
+	Name         string
+	Config       *config.MetadataPlugin
+	Tempfile     string
```

Comment:
Sure. I renamed to Cachefile because the file caches the metadata locally and can be purged away. https://github.com/mackerelio/mackerel-agent/pull/333/commits/68e973fcb9c079abb62ae32d0e87ea982e9df43e